From e8e6547988319d7bfbdc428ce330460ca767ef8b Mon Sep 17 00:00:00 2001
From: Ioana Radulescu <ruxandra.radulescu@nxp.com>
Date: Tue, 22 Aug 2017 15:00:29 +0300
Subject: [PATCH 112/741] staging: fsl-dpaa2/mac: Update ethtool ops

commit ba87b16c8058ba8821173e2cd26aecd3d32ac571 from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

Ethtool ops get/set_settings() are deprecated, so implement
get/set_link_ksettings() instead.

These now call the corresponding phy_ethtool_ksettings_*
generic functions, as the old ones also got deprecated
and removed from the kernel entirely.

Signed-off-by: Ioana Radulescu <ruxandra.radulescu@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/staging/fsl-dpaa2/mac/mac.c | 18 ++++++++++--------
 1 file changed, 10 insertions(+), 8 deletions(-)

diff --git a/drivers/staging/fsl-dpaa2/mac/mac.c b/drivers/staging/fsl-dpaa2/mac/mac.c
index 427acaa09306..98a73c5af005 100644
--- a/drivers/staging/fsl-dpaa2/mac/mac.c
+++ b/drivers/staging/fsl-dpaa2/mac/mac.c
@@ -152,16 +152,18 @@ static int dpaa2_mac_stop(struct net_device *netdev)
 	return 0;
 }
 
-static int dpaa2_mac_get_settings(struct net_device *netdev,
-				  struct ethtool_cmd *cmd)
+static int dpaa2_mac_get_link_ksettings(struct net_device *netdev,
+					struct ethtool_link_ksettings *ks)
 {
-	return phy_ethtool_gset(netdev->phydev, cmd);
+	phy_ethtool_ksettings_get(netdev->phydev, ks);
+
+	return 0;
 }
 
-static int dpaa2_mac_set_settings(struct net_device *netdev,
-				  struct ethtool_cmd *cmd)
+static int dpaa2_mac_set_link_ksettings(struct net_device *netdev,
+					const struct ethtool_link_ksettings *ks)
 {
-	return phy_ethtool_sset(netdev->phydev, cmd);
+	return phy_ethtool_ksettings_set(netdev->phydev, ks);
 }
 
 static void dpaa2_mac_get_stats(struct net_device *netdev,
@@ -319,8 +321,8 @@ static const struct net_device_ops dpaa2_mac_ndo_ops = {
 };
 
 static const struct ethtool_ops dpaa2_mac_ethtool_ops = {
-	.get_settings		= &dpaa2_mac_get_settings,
-	.set_settings		= &dpaa2_mac_set_settings,
+	.get_link_ksettings	= &dpaa2_mac_get_link_ksettings,
+	.set_link_ksettings	= &dpaa2_mac_set_link_ksettings,
 	.get_strings		= &dpaa2_mac_get_strings,
 	.get_ethtool_stats	= &dpaa2_mac_get_ethtool_stats,
 	.get_sset_count		= &dpaa2_mac_get_sset_count,
-- 
2.17.1

